File: 2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.go
func countSubarrays(nums []int, k int64) int64 {
    var result int64 = 0;
    var sum int64 = 0;
    left := 0;

    for right := 0; right < len(nums); right ++ {
        sum += int64(nums[right]);

        for left <= right && sum * int64(right - left + 1) >= k {
            sum -= int64(nums[left]);
            left++;
        }

        result += int64(right - left + 1);
    }

    return result;
}

File: 2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        long result = 0;
        long sum = 0;
        int left = 0;

        for (int right = 0; right < nums.length; right++) {
            sum += nums[right];

            while (left <= right && sum * (right - left + 1) >= k) {
                sum -= nums[left++];
            }

            result += (right - left + 1);
        }

        return result;
    }
}